graphqlapi/src/main/java/io/stargate/graphql/persistence/schemafirst/SchemaSourceDao.java

```diff
@@ -249,4 +250,63 @@ static boolean hasExpectedSchema(Table table) {
     Column contents = table.column(CONTENTS_COLUMN_NAME);
     return contents != null && contents.type() == Column.Type.Varchar;
   }
+
+  public Map<String, Object> tryDeployNewSchema(
+      String namespace, Callable<Map<String, Object>> newSchemaCreator) throws Exception {
+    ensureTableExists(namespace);
+    if (transitionToSchemaDeploymentInProgress(namespace)) {
+      try {
+        return newSchemaCreator.call();
+      } finally {
+        transitionToNoSchemaDeploymentInProgress(namespace);
+      }
+    } else {
+      throw new IllegalStateException(
+          "There is a pending schema deployment process. Please retry later.");
+    }
+  }
+  /**
+   * Returns true if there was no schema deployment in progress. It means that that is safe to
+   * deploy a new schema. If it returns false, it means that there is already schema deployment in
+   * progress.
+   */
+  private boolean transitionToSchemaDeploymentInProgress(String namespace)
+      throws ExecutionException, InterruptedException {
+
+    BuiltCondition schemaDeploymentInProgressNull =
+        BuiltCondition.of(DEPLOYMENT_IN_PROGRESS, Predicate.EQ, null);
+
+    BoundQuery updateDeploymentToInProgress =
+        dataStore
+            .queryBuilder()
+            .update(namespace, TABLE_NAME)
+            .value(DEPLOYMENT_IN_PROGRESS, true)
+            .where(KEY_CONDITION)
+            .ifs(schemaDeploymentInProgressNull)
```

Comment:
We also need to check that the latest version matches:
```suggestion
            .ifs(schemaDeploymentInProgressNull)
            .ifs(LATEST_VERSION_COLUMN_NAME, Predicate.EQ, expectedLatestVersion)
```
(`expectedLatestVersion` must be added as a new method parameter)

This will cover the scenario:
1. user1 reads the schema at version v2
2. user2 reads the schema at v2, then deploys, which creates v3
3. user1 tries to deploy on top of v2

Even though the boolean is back to false, we must not allow this because user1 would override user2's changes.